src/Mod/Draft/draftguitools/gui_dimensions.py

```diff
@@ -504,8 +504,15 @@ def action(self, arg):
                                             self.arctrack.setCenter(self.center)
                                             self.arctrack.on()
                                             for e in self.edges:
-                                                for v in e.Vertexes:
-                                                    self.pts.append(self.arctrack.getAngle(v.Point))
+                                                if e.Length < 0.00003: # Edge must be long enough for the tolerance of 0.00001mm to make sense.
+                                                    _msg(translate("draft", "Edge too short!"))
+                                                    self.finish()
+                                                    return
+                                                for i in [0, 1]:
+                                                    pt = e.Vertexes[i].Point
+                                                    if pt.isEqual(self.center, 0.00001): # A relatively high tolerance is required.
+                                                        pt = e.Vertexes[i - 1].Point     # Use the other point instead.
```

Comment:
Is it acceptable that this will result in the same vertex being used twice?

Reply:
Yes, that is what we also see for an edge that is some distance away from self.center. Both endpoints then produce the same angle. And we need the same to happen when one endpoint (almost matches) self.center.